src/analysis/modals/ComputeModal/GcpComputeModal/GcpComputeImageSelect.ts

```diff
@@ -0,0 +1,107 @@
+import React from 'react';
+import { h } from 'react-hyperscript-helpers';
+import { getImageUrl } from 'src/analysis/modal-utils';
+import { ComputeImage, ComputeImageStore, useComputeImages } from 'src/analysis/useComputeImages';
+import { RuntimeToolLabel, runtimeTools } from 'src/analysis/utils/tool-utils';
+import { spinnerOverlay } from 'src/components/common';
+import { GetRuntimeItem } from 'src/libs/ajax/leonardo/models/runtime-models';
+
+export const defaultImageId = 'terra-jupyter-gatk';
+
+export interface GcpComputeImageSelectProps {
+  readonly setSelectedImage: (image: ComputeImage | undefined) => void;
+  readonly setIsCustomSelectedImage: (value: boolean) => void;
+  readonly tool: RuntimeToolLabel;
+  readonly currentRuntime?: GetRuntimeItem;
```

Comment:
GetRuntimeTime is a pretty big interface/shape... the underlying logic calls getImageUrl which only needs the .runtimeImages property.  I would recommend narrowing the argument type in modal-utils.ts getImageUrl function to only need Pick<GetRuntimeItem, 'runtimeImages'>, and narrow the prop type here to be the same Pick.  That will greatly reduce the burden on mocking in unit tests, and also make the component more flexible since it's needs are narrowed.